content-script.js

```diff
@@ -1,157 +1,254 @@
-            name
-            color
-            description
-            url
+        return `
+          repo${index}: repository(owner: "${owner}", name: "${name}") {
+            nameWithOwner
+            ${info.map((issue, i) => {
+              return `issue${i}: issueOrPullRequest(number: ${issue.id}) { ... on Issue { ...IssueFragment } ... on PullRequest { ...PRFragment } }`
+            })}
           }
+        `
+      })}
+    }
+
+    fragment IssueFragment on Issue {
+      number
+      labels(first: 100) {
+        nodes {
+          name
+          color
+          description
+          url
         }
       }
-      fragment PRFragment on PullRequest {
-        number
-        labels(first: 100) {
-          nodes {
-            name
-            color
-            description
-            url
-          }
+    }
+    fragment PRFragment on PullRequest {
+      number
+      labels(first: 100) {
+        nodes {
+          name
+          color
+          description
+          url
         }
       }
-    `
-    const result = await fetch('https://api.github.com/graphql', {
-      headers: {Authorization: `Bearer ${accessToken}`},
-      method: 'POST',
-      body: JSON.stringify({
-        query: gql
-      })
-    }).then(j => j.json())
-    for (const repoData of Object.values(result.data)) {
-      const repo = repoData.nameWithOwner
+    }
+  `;
+
+  return fetch('https://api.github.com/graphql', {
+    headers: {
+      Authorization: `Bearer ${accessToken}`
+    },
+    method: 'POST',
+    body: JSON.stringify({
+      query: gql
+    })
+  })
+  .then(resp => {
+    if (!resp.ok) {
+      return null;
+    }
+
+    return resp.json();
+  })
+  .then(({ data }) => {
+    const result = {};
+
+    for (const repoData of Object.values(data)) {
+      const repo = repoData.nameWithOwner;
+
       for (const issueData of Object.values(repoData)) {
-        if (typeof issueData === 'string') continue
-        const { number, labels } = issueData
-        cache.set(`${repo}#${number}`, { labels: labels.nodes })
+        if (typeof issueData === 'string') {
+          continue;
+        }
+
+        const { number, labels } = issueData;
+
+        if (labels.nodes.length === 0) {
+          continue;
+        }
+
+        const info = {
+          id: number,
+          labels: labels.nodes
+        }
+
+        if (typeof result[repo] === 'undefined') {
+          result[repo] = [info];
+        } else {
+      result[repo].push(info);
+        }
       }
     }
+
+    return result;
+  });
+}
+
+/**
+* parse label color
+* @param {[number, number, number, number, number, number]} color - Issues/PR label color
+* @returns {[number, number, number, number, number, number]}
+*/
+const parseColor = (() => {
+  return (color) => {
+    const r = +('0x' + color[0] + color[1]);
+    const g = +('0x' + color[2] + color[3]);
+    const b = +('0x' + color[4] + color[5]);
+
+    return [r, g, b, ...rgbToHsl(r, g, b)];
```

Comment:
Imo better to name these:
```suggestion
    return {r, g, b, ...rgbToHsl(r, g, b)};
```